### PR TITLE
remove process.on, use strict, and js files. Update typescript nightly->beta

### PR DESCRIPTION
**Pull request by vishnureddy17**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -1,5 +1,3 @@
-'use strict'
```
Comment (vishnureddy17) [inline src/client.ts:1]: esm is strict by default so no need for this

File: src/connectionFactory/interfaces/webSocketOptions.js [hunk 2]
```diff
@@ -1,2 +0,0 @@
-"use strict";
-exports.__esModule = true;
```
Comment (vishnureddy17) [inline src/connectionFactory/interfaces/webSocketOptions.js:2]: What is this file for? I assume we don't want any JS in our `src` directory

File: src/index.ts [hunk 3]
```diff
@@ -11,18 +11,9 @@ import { UniqueMessageIdProvider } from './uniqueMessageIdProvider.js'
...
 import { URL } from 'url'
 
-process.on("unhandledRejection", (err) => {
-  throw err;
-})
-
-process.on("uncaughtException", (err) => {
-  handle(err);
-})
-
```
Comment (vishnureddy17) [inline src/index.ts:25]: We should not have this. If a user imports our package, then every single uncaught exception or unhandled rejection that happens on that Node process will be altered by us.

**Review by YoDaMa — APPROVED** [review]
lgtm

**Pull request merged**